sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.AddDatePolicy;
+import com.azure.core.http.policy.AddHeadersPolicy;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.http.policy.RequestIdPolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.ai.textanalytics.implementation.TextAnalyticsClientImpl;
+import com.azure.ai.textanalytics.implementation.TextAnalyticsClientImplBuilder;
+import com.azure.ai.textanalytics.models.TextAnalyticsClientOptions;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This class provides a fluent builder API to help instantiation of {@link TextAnalyticsClient TextAnalyticsClients}
+ * and {@link TextAnalyticsAsyncClient TextAnalyticsAsyncClients}, call {@link #buildClient()} buildClient} and
+ * {@link #buildAsyncClient() buildAsyncClient} respectively to construct an instance of the desired client.
+ *
+ * <p>The client needs the service endpoint of the Azure Text Analytics to access the resource service.
+ * {@link #subscriptionKey(String) subscriptionKey(String)} or
+ * {@link #credential(TokenCredential) credential(TokenCredential)} give the builder access credential. </p>
+ *
+ * <p>Another way to construct the client is using a {@link HttpPipeline}. The pipeline gives the client an
+ * authenticated way to communicate with the service. Set the pipeline with {@link #pipeline(HttpPipeline) this} and
+ * set the service endpoint with {@link #endpoint(String) this}. Using a
+ * pipeline requires additional setup but allows for finer control on how the {@link TextAnalyticsClient} and
+ * {@link TextAnalyticsAsyncClient} is built.</p>
+ *
+ * @see TextAnalyticsAsyncClient
+ * @see TextAnalyticsClient
+ */
+@ServiceClientBuilder(serviceClients = {TextAnalyticsAsyncClient.class, TextAnalyticsClient.class})
+public final class TextAnalyticsClientBuilder {
+    private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String CONTENT_TYPE_HEADER_VALUE = "application/json";
```

Comment:
This and the accept header value could be aggregated into CONTENT_TYPE. I don't see any extra value in having two constants that point to the same value.